File: backend/gatorsched_api/schemas/scheduler.py
from datetime import date

from pydantic import BaseModel


class GenerateScheduleRequest(BaseModel):
    date: date


class ScheduledShift(BaseModel):
    id: str
    employeeName: str
    startLabel: str
    endLabel: str
    color: str


class RoleGroup(BaseModel):
    role: str
    shifts: list[ScheduledShift]


class GenerateScheduleResponse(BaseModel):
    groups: list[RoleGroup]

File: frontend/src/services/scheduler.ts
import { API_BASE } from './apiConfig';

export type ScheduledShift = {
  id: string;
  employeeName: string;
  startLabel: string;
  endLabel: string;
  color: string;
};

export type RoleGroup = {
  role: string;
  shifts: ScheduledShift[];
};

export async function generateSchedule(date: string): Promise<RoleGroup[]> {
  const res = await fetch(`${API_BASE}/scheduler/generate`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({ date }),
  });

  if (!res.ok) {
    throw new Error('Failed to generate schedule');
  }

  const data = await res.json();

  return data.groups;
}
